system/lib/dlmalloc.c

```diff
@@ -21,6 +21,10 @@
 /* XXX Emscripten Tracing API. This defines away the code if tracing is disabled. */
 #include <emscripten/trace.h>
 
+#ifdef __EMSCRIPTEN_WASM_WORKERS__
+#define USE_LOCKS 1
```

Comment:
Oh, forgot to mention that this change might be a bit scary, since Wasm Workers do not (currently at least) support the full pthread machinery, which includes the pthread_mutex_*() constructs. Though looks like the test suite doesn't fail at least.

It may be that we'll need to enable `USE_SPIN_LOCKS` as well for Wasm Workers, though I'll defer that kind of analysis for until there is a test case that uncovers an issue here.

Reply:
I had wondered about this as well at the start, but then in my testing it looked like USE_SPIN_LOCKS gets defined automatically on line 661 in dlmalloc.c:
```
#if ((defined(__GNUC__) &&                                              \
((__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 1)) ||      \
defined(__i386__) || defined(__x86_64__))) ||                    \
(defined(_MSC_VER) && _MSC_VER>=1310))
#ifndef USE_SPIN_LOCKS
#define USE_SPIN_LOCKS 1
```
That matches the following comment in the same file:
```
 USE_SPIN_LOCKS           default: 1 iff USE_LOCKS and spin locks available
 If true, uses custom spin locks for locking. This is currently
 supported only gcc >= 4.1, older gccs on x86 platforms, and recent
 MS compilers.  Otherwise, posix locks or win32 critical sections are
 used.
```